felix/bpf/polprog/pol_prog_builder.go

```diff
@@ -406,20 +407,36 @@ func (p *Builder) writeRecordRuleHit(r Rule, skipLabel string) {
 }
 
 func (p *Builder) setUpIPSetKey(ipsetID uint64, keyOffset int16, ipOffset, portOffset FieldOffset) {
+	v6Adjust := int16(0)
+	prefixLen := int32(128)
+
+	if p.forIPv6 {
+		v6Adjust = 12
```

Comment:
Magic number?  Wonder if it'd be cleaner to have a couple of structs defined with the offsets for IPv4 and IPv6, then pick the right one.
```
if ipV6 {
  offsets = v6Offsets
} else {
  ofsets = v4Offsets
}

...
p.b.StoreStack8(R1, offsets.keyOffset+offsets.ipsKeyPad)
```
Or at least a comment explain the fudge factor.